.github/workflows/ci.yml

```diff
@@ -0,0 +1,76 @@
+name: CI
+
+on: pull_request
+
+jobs:
+  setup_matrix:
+    name: 'Setup Test Matrix'
+    runs-on: ubuntu-latest
+    outputs:
+      beaker_setfiles: ${{ steps.get-outputs.outputs.beaker_setfiles }}
+      puppet_major_versions: ${{ steps.get-outputs.outputs.puppet_major_versions }}
+      puppet_unit_test_matrix: ${{ steps.get-outputs.outputs.puppet_unit_test_matrix }}
+    env:
+      BUNDLE_WITHOUT: development:test:release
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '2.7'
+          bundler-cache: true
+      - name: Run rake validate
+        run: bundle exec rake validate
+      - name: Setup Test Matrix
+        id: get-outputs
+        run: bundle exec metadata2gha --use-fqdn --pidfile-workaround false
+
+  unit:
+    needs: setup_matrix
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{fromJson(needs.setup_matrix.outputs.puppet_unit_test_matrix)}}
+    env:
+      BUNDLE_WITHOUT: development:system_tests:release
+      PUPPET_VERSION: "~> ${{ matrix.puppet }}.0"
+    name: Puppet ${{ matrix.puppet }} (Ruby ${{ matrix.ruby }})
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - name: Run tests
+        run: bundle exec rake
+
+  acceptance:
+    needs: setup_matrix
+    runs-on: ubuntu-latest
+    env:
+      BUNDLE_WITHOUT: development:test:release
+    strategy:
+      fail-fast: false
+      matrix:
+        setfile: ${{fromJson(needs.setup_matrix.outputs.beaker_setfiles)}}
+        puppet: ${{fromJson(needs.setup_matrix.outputs.puppet_major_versions)}}
+    name: ${{ matrix.puppet.name }} - ${{ matrix.setfile.name }}
+    steps:
+      - name: Enable IPv6 on docker
+        run: |
+          echo '{"ipv6":true,"fixed-cidr-v6":"2001:db8:1::/64"}' | sudo tee /etc/docker/daemon.json
```

Comment:
My wild guess is that configuring a nonrouteable prefix might cause issues

Reply:
I can test that however Im not sure its what is causing the issue.  The the tests are explicitly failing on the following (and only on some combinations of OS + puppet version)
`dig +tcp +dnssec SOA dnssec-failed.org @127.0.0.1`
but not:
`dig +dnssec +cd SOA dnssec-failed.org @127.0.0.1`

the nsset for dnssec-failed.org also has ipv4 and ipv6 nameservers so broken ipv6 shouldn't affect unbond in this way

Reply:
The fact that the latter command works suggests that unbound is able to fetch all the data it needs to answer the former questions but cant for what ever reason

Reply:
Ok turns out you where right, it seems to take a long time fetching all of the DNSSEC material as it needs to first get the DNSKEY/DS set from the the root, org and dnssec-failed.org zone.  As the data on dnssec-failed.org is bad by design it checks all 6 name-servers. Further because it thinks it has Ipv6 it has to first timeout on the ipv6 address before trying the v4 address. 

the subsequent query works within the time because we dont care about validation so unbound is just sending the bad data from the first thing to respond if its good or not.  i have created https://github.com/voxpupuli/modulesync_config/pull/694